platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/internal/format/ShrinkableSizeCache.java

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.logging.api.internal.format;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Collection;
+import java.util.Deque;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedDeque;
+
+/**
+ * A concurrent map that at given frequency shrinks back to a max size by removing the eldest entries. This map size is
+ * not fixed, it will eventually shrink-back to its maximum.
+ *
+ * @param <K> the type for key
+ * @param <V> the type for value
+ */
+public class ShrinkableSizeCache<K, V> implements Map<K, V> {
+    public static final int MAX_ENTRIES = 10000;
+    public static final int SHRINK_PERIOD_MS = 1000;
+    private final Deque<K> insertionOrderList = new ConcurrentLinkedDeque<>();
+    private final Map<K, V> delegatedMap = new ConcurrentHashMap<>();
+
+    /**
+     * Creates a ShrinkableSizeCache with MAX_ENTRIES max size and SHRINK_PERIOD_MS shrink period.
+     */
+    public ShrinkableSizeCache() {
+        this(ShrinkableSizeCache.MAX_ENTRIES);
+    }
+
+    /**
+     * Creates a ShrinkableSizeCache with configurable max size and shrink period.
+     * @param maxSize configurable max size for the cache
+     * @param shrinkPeriodInMs configurable shrink period for the cache
+     */
+    public ShrinkableSizeCache(final int maxSize, final int shrinkPeriodInMs) {
+        TimerTask cleanUpTask = new TimerTask() {
```

Comment:
Let's not create a single thread here but wait until we have a base project thread.

Reply:
added a different constructor expecting the scheduledExcecutor